cube/src/main/java/zone/cogni/asquare/cube/sort/SkolemizedModel.java

```diff
@@ -0,0 +1,48 @@
+package zone.cogni.asquare.cube.sort;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.ResourceFactory;
+import org.apache.jena.rdf.model.Statement;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.function.Function;
+
+public class SkolemizedModel implements Function<Model, Model> {
+
+  public SkolemizedModel(@Nonnull String skolemBaseUri) {
+    this.skolemBaseUri = skolemBaseUri;
+  }
+
+  private String skolemBaseUri;
+
+  @Override
+  public Model apply(Model model) {
+    Model result = ModelFactory.createDefaultModel();
+    List<Statement> statements = new StatementSorter().apply(model);
+
+    statements.forEach(statement -> {
+      Resource newSubject = statement.getSubject().isURIResource() ? statement.getSubject()
+                                                                   : getSkolemUri(statement.getSubject());
+
+      RDFNode newObject = !statement.getObject().isAnon() ? statement.getObject()
+                                                          : getSkolemUri(statement.getObject());
+
+      Statement newStatement = ResourceFactory.createStatement(newSubject, statement.getPredicate(), newObject);
+      result.add(newStatement);
+    });
+
+    return result;
+  }
+
+  private Resource getSkolemUri(RDFNode blankNode) {
+    return ResourceFactory.createResource(
+            StringUtils.appendIfMissing(skolemBaseUri, "/")
+                    + ".well-known/genid/" + blankNode.asResource().getId().getLabelString()
```

Comment:
Maybe the StringUtils call can go in constructor? Would make this method a little bit more readable (but not a big problem obviously)
Also, I thought you were going to use the hash of the blank node, but this is fine as well for me. :)

Reply:
blank node id here is same as the hash btw